test/integration/cartridge_metrics_test.lua

```diff
@@ -113,3 +113,12 @@ g.test_read_only = function()
     read_only = utils.find_metric('tnt_read_only', resp.json)
     t.assert_equals(read_only[1].value, 1)
 end
+
+g.test_failover = function()
+    helpers.skip_cartridge_version_less('2.7.5')
+    local main_server = g.cluster:server('main')
+
+    local resp = main_server:http_request('get', '/metrics')
+    local failover_trigger_cnt = utils.find_metric('tnt_cartridge_failover_trigger', resp.json)
```

Comment:
Well, it isn't that hard to kill a storage instance and trigger the failover. Maybe we should do this here (you can restart the instance with `after_test` hook).